orca-front50/src/main/groovy/com/netflix/spinnaker/orca/front50/tasks/DeleteDeliveryConfigTask.java

```diff
@@ -38,29 +39,35 @@ public DeleteDeliveryConfigTask(Front50Service front50Service, ObjectMapper obje
   @Nonnull
   @Override
   public TaskResult execute(@Nonnull Stage stage) {
-    if (!stage.getContext().containsKey("deliveryConfigId")) {
+    StageData stageData = stage.mapTo(StageData.class);
+
+    if (stageData.deliveryConfigId == null) {
       throw new IllegalArgumentException("Key 'deliveryConfigId' must be provided.");
     }
 
-    String configId = stage.getContext().get("deliveryConfigId").toString();
-    Optional<DeliveryConfig> config = deliveryConfigUtils.getDeliveryConfig(configId);
+    Optional<DeliveryConfig> config = deliveryConfigUtils.getDeliveryConfig(stageData.deliveryConfigId);
 
     if (!config.isPresent()) {
-      log.debug("Config {} does not exist, considering deletion successful.", configId);
+      log.debug("Config {} does not exist, considering deletion successful.", stageData.deliveryConfigId);
       return new TaskResult(ExecutionStatus.SUCCEEDED);
     }
 
     try {
       log.debug("Deleting delivery config: " + objectMapper.writeValueAsString(config.get()));
     } catch (JsonProcessingException e) {
+      log.warn("Error serializing delivery config object: ", e);
       log.debug("Deleting malformed delivery config:" + config.get());
     }
-    Response response = front50Service.deleteDeliveryConfig(config.get().getApplication(), configId);
+    Response response = front50Service.deleteDeliveryConfig(config.get().getApplication(), stageData.deliveryConfigId);
```

Comment:
`deleteDeliveryConfig()` is going to throw an exception if it's non-200 (or maybe non-200/300) ... general pattern would be if you get a successful response than it's SUCCEEDED otherwise the exceptional case is TERMINAL.